pkg/sql/logictest/testdata/logic_test/schema_change_in_txn

```diff
@@ -1569,3 +1569,67 @@ x  x_b_key  UNIQUE       UNIQUE (b ASC)       true
 
 statement ok
 DROP TABLE x, y
+
+subtest drop_constraint_in_txn
+
+statement ok
+CREATE TABLE t (a INT)
+
+statement ok
+ALTER TABLE t ADD CONSTRAINT c CHECK (a > 0)
+
+statement ok
+BEGIN
+
+statement ok
+ALTER TABLE t DROP CONSTRAINT c
+
+# Since the check constraint is dropped in the schema changer after the
+# transaction commits, it's still enforced during the rest of the transaction.
+statement error pq: failed to satisfy CHECK constraint \(a > 0\)
+INSERT INTO t VALUES (0)
+
+statement ok
+ROLLBACK
+
+statement ok
+ALTER TABLE t ALTER COLUMN a SET NOT NULL
+
+statement ok
+BEGIN
+
+statement ok
+ALTER TABLE t ALTER COLUMN a DROP NOT NULL
+
+# Since the non-null constraint is dropped in the schema changer after the
```

Comment:
This is very astute observation and we should make sure to flag it for docs. You commit already message does, but I'd include the one-line feature summary in the release notes.

Reply:
Done. I also expanded the `schema_change_in_txn` logic tests slightly to cover unvalidated constraints.